docs/source/parsing_best_practices.md

```diff
@@ -0,0 +1,240 @@
+# nvstrings Parsing Best Practices
+
+
+## Regex Performence Tips:
+When working with strings on GPUs, performance is heavily dependent on data and the cuStrings functions you use, especially when doing character/substring substitution and removal.
+
+If you need to write on regex for your pre-processing tasks, follow the below guidelines to achieve best performance.
+
+### Sorting by length:
+
+If your data has unequal length distribution (For example, the longest string might be 1000x longer than the shortest string) and you are doing heavy regex heavy operations, sorting the strings can help to improve the performance.
+
+##### Why:
+
+As regexes operate one character at a time, longer length strings take longer to evaluate and operating on strings in parallel improves throughput.   The GPU works most efficiently when adjacent threads in a warp work in lock step. Wildly different string lengths within adjacent threads means that the threads operating on shorter strings become idle, reducing utilization.
+
+Sorting helps to obtain better wrapping helping adjacent threads in a warp work in lock step to improve utilization . 
+
+### Sorting Alphabetically
+
+Sorting the strings alphabetically can also provide performance gains. 
+
+##### Why:
+
+As each character is processed using a regex, parallel branches may develop causing divergence and reducing efficiency.  Sorting the strings alphabetically can help to reduce this divergence and thus improve the throughput.
+
+
+
+These two tenets of parallel computing mean sorting input strings by length can improve utilization, and sorting the strings alphabetically can improve throughput.
+
+### Example:
+
+Lets walk throw a simple example to see the performance differences for ourselves. 
+
+Lets take a simple example where we have to remove `-` character between 2 alphabetical characters while keeping the `-` between numerical characters on a dataset with 4 million short strings and 50k long strings . 
+
+
+```python
+import random
+import cudf
+
+small_string = "Exmaple-string 123-456-789"
```

Comment:
Exmaple is a tree that used to be maple. I think you mean _Example_ :-)

Reply:
Should be fixed now.  https://github.com/rapidsai/custrings/pull/397/commits/8f5ce58efd95160ba4ff0f2d2014f494da1adb34